hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/impl/ITestS3APutIfMatchAndIfNoneMatch.java

```diff
@@ -390,7 +416,7 @@ public void testIfNoneMatchOverwriteWithEmptyFile() throws Throwable {
 
     // close the stream, should throw RemoteFileChangedException
     RemoteFileChangedException exception = intercept(RemoteFileChangedException.class, stream::close);
-    assertS3ExceptionStatusCode(SC_412_PRECONDITION_FAILED, exception);
+    verifyS3ExceptionStatusCode(SC_412_PRECONDITION_FAILED, exception);
```

Comment:
do you know what the difference is with the other tests here? 

As in, why with S3 express is it ok to assert that we'll get a 412, whereas the others tests will throw a 200?

Reply:
Hey, it's your server code. Go see.

Reply:
checked..the answer is that it's MPU that has divergence, put object which these tests do will return 412